modules/http4s/src/smithy4s/http4s/internals/SmithyHttp4sClientEndpoint.scala

```diff
@@ -102,7 +103,7 @@ private[smithy4s] class SmithyHttp4sClientEndpointImpl[F[_], Op[_, _, _, _, _],
     val baseRequest = Request[F](method, uri, headers = headers)
     if (inputHasBody) {
       baseRequest.withEntity(input)
-    } else baseRequest
+    } else baseRequest.withEntity(ByteVector.empty)
```

Comment:
Should this not only apply to PUT and POST methods?

Reply:
that's a good question, let me see what facilities exists here to distinguish what method allow or not bodies

Reply:
it annoys me that I can't find a public equivalent for this and avoid copy/pasting that method Set out of ember :/

Reply:
> it annoys me that I can't find a public equivalent for this and avoid copy/pasting that method Set out of ember :/

Ah yeah, an issue or PR to fix this would be great. Seems like we just need to add a boolean to `Method`.

Reply:
happy to open it

Reply:
Following the discussions in the [PR](https://github.com/disneystreaming/smithy4s/pull/522), I think this should be merged as in. I'm not sure there can be an always accurate way of telling whether a particular method should have a body or not.

Considering it works for our current use cases, we should move forward with it. If we get and issue later on that sending and empty body is problematic under certain situation, we should revisit.